tests/unit/test_kite_object.py

```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+import pytest
+from mock import patch
+import responses
+
+from kiteconnect import KiteConnect
+import kiteconnect.exceptions as ex
+
+
+def get_fake_token(self, route, params=None):
+    return {"access_token": "TOKEN"}
+
+
+def get_fake_delete(self, route, params=None):
+    return {"message": "token invalidated"}
+
+
+class TestKiteConnectObject:
+
+    def test_login_url(self, kiteconnect):
+        assert kiteconnect.login_url() == 'https://kite.trade/connect/login?api_key=<API-KEY>'
+
+    @responses.activate
+    def test_set_session_expiry_hook_meth(self, kiteconnect):
+
+        def mock_hook():
+            raise ex.TokenException("token expired it seems! please login again")
+
+        kiteconnect.set_session_expiry_hook(mock_hook)
+
+        # Now lets try raising TokenException
+        responses.add(
+            responses.GET,
+            "%s%s" % (kiteconnect.root, kiteconnect._routes["portfolio.positions"]),
```

Comment:
Use `.format` method for string formatting. Pull and check the latest tests.

Reply:
Just found out that `format` is the new way. I assumed the opposite till now. :(